File: app/src/main/java/com/dimatest/movieapp/common/BaseViewHolder.java
package com.dimatest.movieapp.common;

import androidx.annotation.NonNull;
import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;

public abstract class BaseViewHolder<DB extends ViewDataBinding, T> extends RecyclerView.ViewHolder {

    public DB dataBinding;

    public BaseViewHolder(@NonNull DB dataBinding) {
        super(dataBinding.getRoot());
        this.dataBinding = dataBinding;
    }

    public void bind(T item) {
        dataBinding.setVariable(BR.item, item);
    }
}

File: app/src/main/java/com/dimatest/movieapp/common/BaseViewHolder.kt
package com.dimatest.movieapp.common

import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView
import com.dimatest.movieapp.BR

abstract class BaseViewHolder<DB : ViewDataBinding, T>(private val dataBinding: DB) : RecyclerView.ViewHolder(dataBinding.root) {
    open fun bind(item: T) {
        dataBinding.setVariable(BR.item, item)
    }
}